File: iteration2/Lecturer.java
package iteration2;

import java.util.ArrayList;
import java.util.List;

public class Lecturer extends Person {

    List<Course> lessonsTaught;

    //For creating lecturer with person abstract class.
    public Lecturer(int ID, String name, String surname) {
        super(ID, name, surname, null, null);
        lessonsTaught = new ArrayList<>();
    }

    //For creating advisor with person abstract class.
    public Lecturer(int ID, String name, String surname, String userName, String password) {
        super(ID, name, surname, userName, password);
        lessonsTaught = new ArrayList<>();
    }

    //Get the given courses lecturer gives.
    public List<Course> getLessonsTaught() {
        return lessonsTaught;
    }

    //Advisor login proccess.
    @Override
    public boolean login(String userName, String password) {
        return false;
    }
}

File: iteration3/Lecturer.py
from Person import Person


class Lecturer(Person):
    # For creating lecturer with person abstract class.
    def __init__(self, ID, name, surname, userName, password):
        super().__init__(ID, name, surname, userName, password)
        self.lessonsTaught = []

    # Get the given courses lecturer gives.
    def getLessonsTaught(self):
        return self.lessonsTaught

    # Advisor login process.
    def login(self, userName, password):
        return False
